README.md

```diff
@@ -1,3 +1,202 @@
-# UNDER DEVELOPMENT
-# r16n
-I18n for Redux
+<img src="https://cdn.rawgit.com/razodeh/r16n/ab1ec653/logo/logo-title.png" height="50px" alt="R16N Logo" /> 
+
+# Reduxtionalization (Redux I18N) [![Build Status](https://travis-ci.org/razodeh/r16n.svg?branch=master)](https://travis-ci.org/razodeh/r16n) 
+
+[Installation](#installation) | [Usage](#usage) | [Bindings](#bindings)
+
+## Installation
+Using npm:
+```
+npm install r16n
+```
+Using yarn:
+```
+yarn add r16n
+```
+
+## Usage
+
+### 1. Create R16N Reducer
+Create R16N reducer with your own locales, and the current locale.
+
+```javascript
+import {createR16nReducer} from 'r16n';
+// ...
+const locales = {
+    en: {
+        helloWorld: 'Hello, World!',
+        navbar: {
+	            home: 'Home'
+        }
+    },
+    ar: {
+        helloWorld: 'مرحباً أيها العالم',
+        navbar: {
+            home: "الرئيسية"
+        }
+    },
+    zh_HK: {
+		helloWorld: "你好世界",
+		navbar: {
+			home: "主页",
+		}
+	}
+};
+// Your default locale, whether it is fixed, or imported from a cookie or an API.
+const currentLocale = 'en';
+const reducers = combineReducers({
+	// ...
+    r16n: createR16nReducer(locales, currentLocale),
+});
+```
+> R16N's reducer name has to be `r16n`, otherwise it won't work.
+
+### 2. Select Your Translations
+
+#### `getTranslation(state, key)`:
+Returns the translation state of the requested key, and count.
+**args** :
+
+ - `state` : Redux store state.
+ - `key` : dot-separated string indicating your translation path.
+
+If your locales object was the following:
+```javascript
+{
+	en:{
+		dummyKey: 'Dummy Translation',
+		nestedKey: {
+			anotherKey: 'Another Translation'
+		}
+	},
+	// ... other locales.
+}
+```
+`dummyKey`, and `nestedKey.anotherKey` are valid translation keys.
+
+**example** :
+```javascript
+import {getTranslation} from 'r16n';
+// ...
+const mapStateToProps = (state) => ({
+	// ...
+    appleText: getTranslation(state, 'navbar.home'), // en-> Home, ar-> الرئيسية
+});
+export default connect(mapStateToProps, ...)(/*Your Component*/);
+```
+> The same works on every locale you provide.
+
+#### `getTranslations(state)`:
```

Comment:
These are better to be placed in numbered subsections like this:

#### 2.2 Getting the Current translation object
To get the current translations object in case you needed it, call  `getTranslations(state)`...